### ISSUE TITLE
Date (skeletons): dynamically augment availableFormats

### ISSUE THREAD

**Issue by rxaviers** [globalizejs/globalize#271]
WIP PR: #703 (replaced  ~~#604~~  and ~~#462~~)

---

As specification says:

> It is not necessary to supply dateFormatItems with skeletons for every field length; in many cases fields in the skeleton and pattern can be expanded in parallel to handle a request. Consider the following dateFormatItem:
> 
> `<dateFormatItem id="yMMMd">d MMM y</dateFormatItem>`
> 
> If the request is for yMMMMd, the above item can be automatically expanded to produce the pattern "d MMMM y" in response to the request. Of course, if the desired behavior is that a request for yMMMMd should produce something other than "d MMMM y", a separate dateFormatItem must be added, for example:
> 
> `<dateFormatItem id="yMMMMd">d 'de' MMMM 'de' y</dateFormatItem>`

http://www.unicode.org/reports/tr35/tr35-dates.html#availableFormats_appendItems

We need to support that on `.formatDate()`.

**Comment by AndreR** [comment]
Any news on this topic? I wanted to use the skeleton "MMMM" with Globalize set to German, but I ended up with that exact same error message _E_UNSUPPORTED: Unsupported {"feature": "year pattern `u`"}_. If I set it to English, it works, though.

Any chance that this will be fixed anytime soon?

**Comment by rxaviers** [comment]
Hi @AndreR, #604 is stalled. Would you like to take it over? I'd be happy if you could address the comments in that PR. You could create a new pull request on top of that one with additional commits. Just let me know if you have any questions. Thanks

**Comment by AndreR** [comment]
Hi @rxaviers, I would love to help you out here, but my Javascript skills are too basic. I'm sorry!
